DSharpPlus/Exceptions/DiscordException.cs

```diff
@@ -24,27 +24,29 @@
 using System;
 using DSharpPlus.Net;
 
-namespace DSharpPlus.Exceptions
+namespace DSharpPlus.Exceptions;
+
+public abstract class DiscordException : Exception
 {
-    public abstract class DiscordException : Exception
-    {
-        /// <summary>
-        /// Gets the request that caused the exception.
-        /// </summary>
-        public virtual BaseRestRequest WebRequest { get; internal set; }
+    /// <summary>
+    /// Gets the request that caused the exception.
+    /// </summary>
+    public virtual BaseRestRequest WebRequest { get; internal set; } = null!;
+
+    /// <summary>
+    /// Gets the response to the request.
+    /// </summary>
+    public virtual RestResponse WebResponse { get; internal set; } = null!;
 
-        /// <summary>
-        /// Gets the response to the request.
-        /// </summary>
-        public virtual RestResponse WebResponse { get; internal set; }
+    /// <summary>
+    /// Gets the JSON message received.
+    /// </summary>
+    public virtual string? JsonMessage { get; internal set; } = null!;
 
-        /// <summary>
-        /// Gets the JSON message received.
-        /// </summary>
-        public virtual string JsonMessage { get; internal set; }
+    /// <inheritdoc />
+    public override string Message => $"{base.Message}. Json Message: {JsonMessage ?? "No Json Message Provided."}";
```

Comment:
this really does not need to be in title case; and i think we should come up with a better name than JsonMessage. something like "Discord did not provide an error message."?